### PR TITLE
Add calculation of PU usage for each run

### PR DESCRIPTION
**Pull request by ikas**
### Overview

This PR adds the calculation of PU usage on a per run basis to the parsing of calculation results.

### Feature relevant tickets

[MARXAN-810](https://vizzuality.atlassian.net/browse/MARXAN-810)

### CODE REVIEW

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/solutions-output/result-parser.service.ts [hunk 1]
```diff
@@ -81,46 +75,12 @@ export class ResultParserService {
...
-              // `MostDifferentService`.
-              distinctFive: false,
-              puValues: planningUnitSelection['1'].values.map((b) =>
-                b ? 1 : 0,
-              ),
-            });
+
+            const entry = ResultParserService.calculatePuValues(
+              ResultParserService.parseRowToResultRow(row),
+              planningUnitSelection,
```
Comment (kgajowy) [inline api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/solutions-output/result-parser.service.ts:81]: I may be not noticing something obvious - sorry 😬 - is there any reason why we cannot "just" assign that value within `parseRowToResultRow`? If possible, would be a bit more straightforward to understand.

**Comment by kgajowy** [comment]
Also:
image

**Comment by kgajowy** [comment]
E2E on real marxan run seems to fail as well during parsing:
https://github.com/Vizzuality/marxan-cloud/runs/4250556227?check_suite_focus=true#step:4:22285

File: api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/solutions-output/result-parser.service.ts (hunk 1 above)
Comment (ikas) [reply api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/solutions-output/result-parser.service.ts:81]: I was aiming at maintaining the existing behavior, where we only set `puValues` in the second call of this function (line 120 of the old file), whereas in the initial call (line 64 of the old file), `puValues` was initialized as empty - but I agree this way doing this is more confusing :upside_down_face:

I'll apply your suggestion and I'll set `puValues` straight away on `parseRowToResultRow` and check the tests - if nothing breaks, we should be ok doing this :) (unless ofc you think we should keep the previous behavior!)

**Comment by ikas** [comment]
Feedback has been addressed, will merge the PRs and have a look at why tests are failing in the target PR (#629)

**Pull request merged**